torch/_inductor/scheduler.py

```diff
@@ -4220,27 +4221,44 @@ def can_buffer_be_removed_through_fusion(
             and name not in self.mutation_real_name
         )
 
-    def should_partition(self, node: BaseSchedulerNode) -> bool:
+    def should_partition(
+        self, node: BaseSchedulerNode, should_log: bool = False
+    ) -> bool:
         """Return True if we should partition the inductor graph on this node"""
+
+        # avoid duplicating logs when should_partition is called multiple times
+        # on the same node
+        def noop_log(msg: str) -> None:
+            return
+
+        log_partition_reason = maybe_log_cudagraph_partition if should_log else noop_log
+
         if isinstance(node, FusedSchedulerNode):
             return any(self.should_partition(snode) for snode in node.snodes)
 
         if not node.is_gpu():
+            log_partition_reason("non gpu ops")
+
             return True
 
         if node.node is None:
+            log_partition_reason("None ops")
             return True
```

Comment:
Why do we skip when node.node is None ? when would this even happen?

Reply:
Oh it was added before [FusedSchedulerNode is handled properly](https://github.com/pytorch/pytorch/pull/147038/commits/c70c46aa12464296a8eed0eaf0877ab1a3ef1cca), which have `node.node` as None and actual node in `node.snodes`.  For lint reasons, we can replace it with `assert node.node is not None` now.